### DISCUSSION TITLE
Can TS Muxer create 4K Blu-rays?

### DISCUSSION THREAD

**Discussion by sebarious70** [justdan96/tsMuxer#888] [Q&A]
I think the answer is probably no, but provided that I import into TS Muxer an UHD video with a Dolby Atmos track, or any audio track, like PCM or AC3, and I set it to build an ISO then burn it, does TS Muxer include in the metadata that it's a 4K Blu-ray so that the 4K Blu-ray player can identify it properly, or does TS Muxer only builds plain HD Blu-rays?

**Comment by donnafaye0323** [comment]
No. it only creates 720p and 1080i Blu-ray videos and AC3 audio according to the Blu-ray Disc Association standards. These are Blu-ray compliant video rules.

**Comment by donnafaye0323** [comment]
No. it only creates 720p and 1080i Blu-ray videos and AC3 audio according to the Blu-ray Disc Association standards. These are Blu-ray compliant video rules. Also, 4K videos are not supported.

**Comment by jcdr428** [comment]
@sebarious70 yes it does create 4k blu-rays.

**Reply by sebarious70** [reply]
Thanks for the clarification. Speaking of which, I have a problem that I hope you or someone else here can help me with. 

About two years ago, I was able to author a 4K Blu-ray in both BD-R 25 and 50, even with HDR metadata, that was properly read by the players I have. I remember doing that with two programs, meaning one version with each, one was TS Muxer and the other one was BD Rebuilder, which also uses TS Muxer. This was home videos edited in FCPX and exported as HDR following all the steps in the FCPX manual.

In the past few days, I edited a video I shot recently in DaVinci Resolve Studio, as an HDR project. This program can render to h.265 with the HDR flag and all that, but it only has two encoders (without paying for others), a native one and an Nvidia one. After doing several tests, I'm pretty sure that none of them produces an encode that is compatible with the 4K Blu-ray spec. 

But when I did the tests that did work two years ago, I remember the previous to last step was to use Handbrake using x265 and making sure to use the parameter --uhd-bd. The encoded file could be imported into BD Rebuilder, which produced a disc that played perfectly even in my Panasonic 4K Blu-ray player, which is the most picky and hardly plays anything that is not 100% spec compliant.

So this time I figured I would export a lossless file to bring into Handbrake and try to replicate the process. I couldn't find the notes from two years ago, but I have been encoding videos for over twenty years, so I knew how to setup all the different parameters.

Now, the file that it produced, when I bring it into the most current build of TS Muxer, it creates the Blu-ray ISO just fine, and after burning it with ImgBurn, it does play in my Sony player, but not perfectly. As long as I don't touch it, it works, but when I fast forward or rewind, sometimes it stalls. In my Panasonic player, when it loads the disc, it stays at 0:00:00 for about thirty seconds until it stops.

So I wanted to try BD Rebuilder because with it I can also add basic menus. It uses TS Muxer for part of the process, but the same file that in TS Muxer created the ISO without problems, in BD Rebuilder always gives me this error:

![Screenshot 2025-03-08 155523](https://github.com/user-attachments/assets/84bf447f-9f75-4b62-b034-d19b03bebf58)

The log window shifts to the right for some reason, but basically it shows it's a TS Muxer Error 0xFFFFFFFE, which it says "Source issue found, attempting correction, and after a few seconds, 0XC0000005 and it fails to build the structure.

Do these error codes mean anything to you? Is there perhaps another parameter besides --uhd-bd that I can insert in Handbrake for x265 that would take care of this problem?

One thing I almost left out, I could tell that the latest version of BD Rebuilder, which is 0.62.12 (August 18, 2024) still has the TSMuxer.exe from like 2021, so searching online I learned that to use the most current TS Muxer I have to copy the exe to the tools folder in BD Rebuilder, then rename it to tsmuxer4k.exe. I did that after the steps described above, but nothing changed, still having the same errors.
